Sources/AsyncHTTPClient/Utils.swift

```diff
@@ -15,6 +15,32 @@
 import NIO
 import NIOHTTP1
 
+#if canImport(Network)
+    import Network
+
+    internal extension String {
+        var isIPAddress: Bool {
+            if IPv4Address(self) != nil || IPv6Address(self) != nil {
+                return true
+            }
+            return false
+        }
+    }
+
+#else
+    internal extension String {
+        var isIPAddress: Bool {
```

Comment:
@Lukasa should we add smth like this to NIO? This is unrelated to this PR but we could make a 'good-first-issue'

Reply:
Yeah, I wouldn't object, though we'd be extending a type we don't own so we'd have to prefix the method name, which is kinda gross.

Reply:
And actually don't we have this already in the form of `SocketAddress(ipAddress:port:)`?

Reply:
@Lukasa ish, need to give a port too. But maybe that's not too bad?

```
extension String {
    var isIPAddress: Bool {
        return SocketAddress(ipAddress: self, port: 0) != nil
   }
}
```

?

Reply:
I suspect it's fine, and the only reason not to use it here is that you do have to heap-allocate for that code.